tests/test_diffusion.py

```diff
@@ -0,0 +1,188 @@
+def Send2PreviousPoint(particle):
+    # print("Recovery triggered at %s | %s!" % (particle.lon, particle.lat))
+    # print("Moving particle back to %s | %s" % (particle.prev_lon, particle.prev_lat))
+    particle.lon = particle.prev_lon
+    particle.lat = particle.prev_lat
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def diffusion_test(mode, type='true_diffusion', particles=1000, timesteps=1000, output_file='diffusion_test'):
+    # Generating grid with zero horizontal velocities
+    forcing_fields = CreateDummyUV(100, 100)
+    grid = Grid(forcing_fields['U'], forcing_fields['V'], forcing_fields['U'].depth,
+                forcing_fields['U'].time, fields=forcing_fields)
+    # Create a non-uniform field of diffusivity
+    grid.add_field(CreateDiffusionField(grid))
+    # Calculate first differential of diffusivity field (needed for diffusion)
+    divK = grid.K.gradient()
+    grid.add_field(divK[0])
+    grid.add_field(divK[1])
+    # Calculate second differential (needed to estimate the appropriate minimum timestep to approximate Eulerian diffusion)
+    grid.add_field(grid.dK_dx.gradient(name="d2K")[0])
+    grid.add_field(grid.dK_dy.gradient(name="d2K")[1])
+
+    # Evenly spread starting distribution
+    Start_Field = Field('Start', CreateStartField(grid.U.lon, grid.U.lat),
+                        grid.U.lon, grid.U.lat, depth=grid.U.depth, time=grid.U.time, transpose=True)
+
+    timestep = 500
+
+    steps = timesteps
+
+    ParticleClass = JITParticle if mode == 'jit' else ScipyParticle
+
+    # Simply particle superclass that remembers previous positions for kernel error recovery
+    class Diffuser(ParticleClass):
+        prev_lon = Variable("prev_lon", to_write=False, initial=attrgetter('lon'))
+        prev_lat = Variable("prev_lat", to_write=False, initial=attrgetter('lat'))
+        new_lon = Variable("new_lon", to_write=False, initial=attrgetter('lon'))
+        new_lat = Variable("new_lat", to_write=False, initial=attrgetter('lat'))
+
+    diffusers = grid.ParticleSet(size=particles, pclass=Diffuser, start_field=Start_Field)
+
+    # Particle density at simulation start should be more or less uniform
+    DensityField = Field('temp', np.zeros((5, 5), dtype=np.float32),
+                         np.linspace(np.min(grid.U.lon), np.max(grid.U.lon), 5, dtype=np.float32),
+                         np.linspace(np.min(grid.U.lat), np.max(grid.U.lat), 5, dtype=np.float32))
+    StartDensity = diffusers.density(DensityField, area_scale=False, relative=True)
+    grid.add_field(Field(type + 'StartDensity', StartDensity,
+                         DensityField.lon,
+                         DensityField.lat))
+
+    diffuse = diffusers.Kernel(SaptiallyVaryingDiffusion2D) if type == 'true_diffusion' else diffusers.Kernel(LagrangianDiffusionNoCorrection)
+
+    diffusers.execute(diffusers.Kernel(UpdatePosition) + diffuse, endtime=grid.U.time[0]+timestep*steps, dt=timestep,
```

Comment:
Since `steps = timesteps` above, why not simply use `timesteps` here, and get rid of the `steps = timesteps` above?

Also, it is probably easier here to use the `runtime` keyword (i.e. `runtime=timesteps*steps`), which sets `endtime` as `starttime + runtime`